src/sentry/models/groupinbox.py

```diff
@@ -29,3 +29,16 @@ class GroupInbox(Model):
     class Meta:
         app_label = "sentry"
         db_table = "sentry_groupinbox"
+
+
+def add_group_to_inbox(group, reason, reason_details):
+    if not GroupInbox.objects.filter(group=group).exists():
+        return GroupInbox.objects.create(
+            group=group, reason=reason.value, reason_details=reason_details
+        )
+    else:
+        return None
```

Comment:
Could potentially just use `get_or_create` here.

Reply:
Hmm true...was concerned about the possibility of this being called with a different reason when it's already in the inbox